src/openeo_test_suite/lib/compliance_util.py

```diff
@@ -0,0 +1,532 @@
+            print("")
+            print(error.message)
+            print(error.json_path)
+    else:
+        print(e)
+    print("------------------------------------------")
+    print("")
+
+
+def check_test_results(e: Exception):
+    """
+    prints the results of a openapi-core validation test
+
+    e: the exception that was raised as a part of the response validation test
+    test_name: the name of the test that ought to be displayed
+    """
+
+    # message is important
+
+    fail_log = ""
+    if type(e) is UnexpectedStatusCodeException:
+        if e.auth and (e.actual_status_code == 500):
+            fail_log = "Bearer-Token invalid/no longer valid or if endpoint expects an id the item does not exist."
+        elif e.actual_status_code == 500:
+            fail_log = "Endpoint expects an id and the item does not exist."
+        elif e.actual_status_code == 404 or e.actual_status_code == 501:
+            fail_log = "Endpoint is not implemented, only an error if it is REQUIRED."
+        elif e.actual_status_code == 410:
+            fail_log = "Endpoint is not providing requested resource as it is gone. Logs are not provided if job is queued or created."
+        else:
+            fail_log = f"Some other unexpected error code. {e.actual_status_code}"
+    # json_path and message are important
+    elif hasattr(e.__cause__, "schema_errors"):
+        errors = e.__cause__.schema_errors
+        for error in errors:
+            fail_log += f"Message: {error.message} Json_path: {error.json_path} \n"
+    else:
+        fail_log = str(e)
+
+    return fail_log
+
+
+def adjust_server(path_to_file, endpoint):
+    with open(path_to_file, "r") as file:
+        data = yaml.safe_load(file)
+
+    if "servers" in data and isinstance(data["servers"], list):
+        for server in data["servers"]:
+            if "url" in server and isinstance(server["url"], str):
+                server["url"] = endpoint
+    return data
+
+
+def adjust_spec(path_to_file: str, endpoint: str, domain: str):
+    data = adjust_server(path_to_file=path_to_file, endpoint=endpoint)
+    data = adjust_server_in_well_known(data=data, endpoint=domain)
+    return Spec.from_dict(data, validator=None)
+
+
+def adjust_server_in_well_known(data, endpoint):
+    data["paths"]["/.well-known/openeo"]["get"]["servers"][0]["url"] = endpoint
+
+    return data
+
+
+def validate_uri(value):
+    return True
+
+
+extra_format_validators = {
+    "uri": validate_uri,
+}
+
+
+def unmarshal_commonmark(value):
+    return value
+
+
+extra_format_unmarshallers = {
+    "commonmark": unmarshal_commonmark,
+}
+
+
+def get_examples_path():
+    return os.path.join(
+        os.getcwd(), "src/openeo_test_suite/tests/general/payload_examples"
+    )
+
+
+def get_spec_path():
+    return os.path.join(
+        os.getcwd(), "src/openeo_test_suite/tests/general/openeo-api/openapi.yaml"
+    )
+
+
+def load_payloads_from_directory(directory_path: str) -> list[dict]:
```

Comment:
return type hint is wrong I think: there is yield, so it's not a list, but an `Iterator` and it yields strings, not dicts

Reply:
Fixed.